### PR TITLE
Run included suites as subtests

### PR DESCRIPTION
**Pull request by artem-belov**

### CODE REVIEW

File: internal/generator/suite.go [hunk 1]
```diff
@@ -40,9 +41,28 @@ func (s *Suite) SetupSuite() {
 	{{ end }}
 	{{ .Cleanup }}
 	{{ .Run }}
+
+	s.RunIncludedSuites()
+}
+
+func (s *Suite) RunIncludedSuites() {
+	{{ .TestIncludedSuites }}
```
Comment (denis-tingaikin) [inline internal/generator/suite.go:49]: We should not generate this if there are no included suits. Consider looking at go-template conditions.

File: internal/generator/suite.go [hunk 2]
```diff
@@ -40,9 +41,28 @@ func (s *Suite) SetupSuite() {
...
 }
 `
 
+const includedSuiteTemplate = `
+	{{ range .Suites }}
+		s.Run("{{ .Title }}", func() {
+			{{ $suiteName := .Name }}
+			s.{{ $suiteName }}Suite.SetT(s.T())
```
Comment (denis-tingaikin) [inline internal/generator/suite.go:57]: Should we call BeforeTest and AfterTest if they are presented?

File: internal/linker/linked_example.go [hunk 3]
```diff
@@ -66,13 +65,20 @@ func (e *LinkedExample) Dependencies() []string {
 		deps = append(deps, child.Name)
 	}
 
+	return append(e.SetupDependencies(), deps...)
+}
+
+// SetupDependencies returns setup dependecies for this example
+func (e *LinkedExample) SetupDependencies() []string {
```
Comment (denis-tingaikin) [inline internal/linker/linked_example.go:72]: SetupDependencies confuses a bit. Consider renaming this.

File: internal/generator/generator.go [hunk 4]
```diff
@@ -60,21 +61,34 @@ func (g *Generator) Generate(examples ...*linker.LinkedExample) []*Suite {
 		var deps = Dependencies([]Dependency{Dependency(g.conf.BasePkg)})
 		deps = append(deps, normalizeDeps(moduleName, e.Dependencies())...)
 
-		result = append(result, &Suite{
+		var setupDeps = Dependencies([]Dependency{Dependency(g.conf.BasePkg)})
```
Comment (denis-tingaikin) [inline internal/generator/generator.go:64]: Basically, to simplify this we can add into the example some function component `IsSubSuite` that will return true if one of the parents includes this example and false for other cases.

So my point to have here a clean section with branches for `tests`, `suites`, and `subsuits`.

**Review by denis-tingaikin — CHANGES_REQUESTED** [review]
In general, I like this patch, It fixes a few major issues. 

Added comments.

File: internal/generator/suite.go (hunk 1 above)
Comment (artem-belov) [reply internal/generator/suite.go]: Done

File: internal/linker/linked_example.go (hunk 3 above)
Comment (artem-belov) [reply internal/linker/linked_example.go]: Done

File: internal/generator/suite.go (hunk 2 above)
Comment (artem-belov) [reply internal/generator/suite.go:86]: Done

File: internal/generator/generator.go (hunk 4 above)
Comment (artem-belov) [reply internal/generator/generator.go]: subSuite and suite has the only difference in containing parent suite. I just added more comments to the code.

**Review by denis-tingaikin — APPROVED** [review]

**Pull request merged**